content/blog/2024-07-10-unikraft-gsoc-lxboot_x86.mdx

```diff
@@ -0,0 +1,80 @@
+---
+title: "GSoC'24: Linux x86 Boot Protocol Support"
+description: |
+  The Linux boot protocol plays an important role in the initialization of the Linux operating system, emphasizing the importance of system optimization and scalability.
+publishedDate: 2024-07-10
+image: /images/unikraft-gsoc24.png
+tags:
+- gsoc
+- gsoc24
+- booting
+authors:
+- Mihnea Firoiu
+---
+
+As I wrote in the previous blog post, my first objective is to make Unikraft boot on QEMU using the Linux x86 boot protocol.
+Here I will present my progress so far and different challenges I faced while working towards my goal.
+
+## Where does QEMU handle the lxboot header?
+
+By using the `git grep` command, I found that `QEMU 9.0.1` handles the Linux x86 boot protocol in the [x86.c](https://github.com/qemu/qemu/blob/stable-9.0/hw/i386/x86.c) file.
+Everything happens in the `x86_load_linux` function.
+
+### Fields used by QEMU
+
+These are as follows:
+
+* 0x202: HEADER
+* 0x206: VERSION
+* 0x211: LOADFLAGS
+* 0x236: XLOADFLAGS
+* 0x22c: INITRD_ADDR_MAX
+* 0x228: CMD_LINE_PTR
+* 0x1fa: VID_MODE
+* 0x210: TYPE_OF_LOADER
+* 0x224: HEAP_END_PTR
+* 0x1f1: SETUP_SECTS
+* 0x250: SETUP_DATA
+
+As you can see, only 11 out of 39 fields are used.
+Out of these, 6 are read by QEMU:
+
+* 0x202: HEADER
+* 0x206: VERSION
+* 0x211: LOADFLAGS
+* 0x236: XLOADFLAGS
+* 0x22c: INITRD_ADDR_MAX
+* 0x1f1: SETUP_SECTS
+
+And 5 are written by QEMU:
+
+* 0x228: CMD_LINE_PTR
```

Comment:
What about [`ramdisk_image`/`ramdisk_size`](https://github.com/qemu/qemu/blob/master/hw/i386/x86-common.c#L879-L880)?

Reply:
I just made the change, I don't know how I forgot to add them.